tests/integration_tests/ibm/test_launch.py

```diff
@@ -0,0 +1,68 @@
+import os
+import pytest
+from pycloudlib.ibm.cloud import IBM
+from pycloudlib.ibm.instance import IBMInstance
+from google.cloud import compute_v1
+import time
+
+
+@pytest.fixture
+def ibm_cloud():
+    with IBM(tag="integration-test-launch") as ibm:
+        yield ibm
+
+
+def manage_ssh_key(ibm: IBM, key_name):
+    """Manage ssh keys for ibm instances."""
+    if key_name in ibm.list_keys():
+        ibm.delete_key(key_name)
+
+    pub_key_path = "ibm-pubkey"
+    priv_key_path = "ibm-privkey"
+    pub_key, priv_key = ibm.create_key_pair()
+
+    with open(pub_key_path, "w", encoding="utf-8") as f:
+        f.write(pub_key)
+
+    with open(priv_key_path, "w", encoding="utf-8") as f:
+        f.write(priv_key)
+
+    os.chmod(pub_key_path, 0o600)
+    os.chmod(priv_key_path, 0o600)
+
+    ibm.use_key(
+        public_key_path=pub_key_path,
+        private_key_path=priv_key_path,
+        name=key_name,
+    )
+
+
+def test_ibm_launch(ibm_cloud: IBM):
+    """
+    Test launching an IBM instance.
+
+    This tests the following:
+    - The instance is launched successfully
+    - The instance is reachable via SSH
+    - The instance has a floating IP as expected
+    - The instance name was set correctly
+    """
+    time_id = time.time_ns()
+    image_id = ibm_cloud.released_image("noble")
+    floating_ip_substring = "default-floating-ip"
```

Comment:
Hard-coding this string in integration tests won't work in most environments unless an available prerequisite floating-ip has already been allocated for a group named with the default-floating-ip prefix (and not already used by some other CI runner like jenkins). 

This test should either, setup it's own unique floating IP with a custom prefix, or source pycloudlib.toml config.

Let's go the pycloudlib.toml config front here for the moment and to proposed diff
```diff
--- a/tests/integration_tests/ibm/test_launch.py
+++ b/tests/integration_tests/ibm/test_launch.py
@@ -49,7 +49,9 @@ def test_ibm_launch(ibm_cloud: IBM):
     """
     time_id = time.time_ns()
     image_id = ibm_cloud.released_image("noble")
-    floating_ip_substring = "default-floating-ip"
+    floating_ip_substring = ibm_cloud.config.get("floating_ip_substring", "")
+    if not floating_ip_substring:
+        floating_ip_substring = "default-floating-ip"
     unique_instance_name = f"integration-test-launch-instance-{time_id}"
     # create ssh_keys for use
     manage_ssh_key(ibm_cloud, f"integration-test-key-{time_id}")
```
```

Also, i'm not a big fan of our generic IBMException for any IBM error (not due to this PR). We should specialize errors so we know what to report to end-users.  In server team case, we happen to name our floating ip's differently than the `"default-floating-ip`, we I get the following traceback from integration tests:
```
        if not floating_ips:
>           raise IBMException(
                f"No floating IPs found matching substring {name_includes}"
            )
E           pycloudlib.ibm.errors.IBMException: No floating IPs found matching substring default-floating-ip
```

For a separate PR: what I'd rather see is a raised exception that is a subclass of pycloudlib.errors.ResourceNotFoundError  (maybe named `FloatingIPNotFoundError`). Then I can have a try/except FloatingIPNotFoundError around the launch to better inform the integration-test runner that they haven't pre-allocated the expected availble floating-ips in order to run the test


That comment about subclassing ResourceNotFoundErrors in IBM applies to many of the errors generated by pycloudlib.ibm.instance which can probably be taken to a separate issue/PR
 - "Subnet not found" 
 - No subnet associated to vpc found
 - VPC not found
 - No subnet available

Reply:
@blackboxsw 100% agree with the generic exception feedback. Will create an issue for this. Has been on the back of my mind for a while.